src/metrics/mod.rs

```diff
@@ -97,9 +97,9 @@ impl HistogramsSnapshot {
                 continue;
             };
 
-            if let Some(histogram) = any.downcast_ref::<metriken::AtomicHistogram>() {
-                if let Some(snapshot) = histogram.snapshot() {
-                    current.insert(metric.name().to_string(), snapshot);
+            if let Some(atomic_histogram) = any.downcast_ref::<metriken::AtomicHistogram>() {
```

Comment:
I wonder if .map(|h| h.load()).flatten() would be cleaner here

Reply:
Moved to using `and_then` instead